contrib/julia-mode.el

```diff
@@ -131,6 +131,23 @@ This function provides equivalent functionality, but makes no efforts to optimis
              "\\\\"))
         (group "'"))))
 
+(defconst julia-hanging-assignment-regexp
+  (concat "^[^#\n]+ "
+          (regexp-opt '("=" "+=" "-=" "*=" "/=" "//=" "\\=" "^=" "÷=" "%=" "|=" "&=" "$=" "<<=" ">>="
+                        ".+=" ".-=" ".*=" "./=" ".//=" ".\\=" ".^=" ".÷=" ".%="))
+          (regexp-opt '(" #" " \n" "#" "\n"))))
+
+(defconst julia-hanging-operator-regexp
+  (concat "^[^#\n]+ "
+          (regexp-opt '("^" ".^" "//" ".//" "*" "/" "%" "&" "\\" ".*" "./" ".%" ".\\"
+                        "<<" ">>" ">>>" ".<<" ".>>" ".>>>"
+                        "+" "-" "|" "$" ".+" ".-"
+                        ":" ".." "|>"
+                        ">" "<" ">=" "<=" "==" "===" "!=" "!==" "<:" "≠" "≤" "≥"
+                        ".>" ".<" ".>=" ".<=" ".==" ".!="
+                        "&&" "||" "?"))
+          (regexp-opt '(" #" " \n" "#" "\n"))))
```

Comment:
Ok thanks. I'll look into it more tomorrow
On Fri, Feb 19, 2016 at 10:45 PM Yichao Yu <notifications@github.com> wrote:

> In contrib/julia-mode.el
> <https://github.com/JuliaLang/julia/pull/15156#discussion_r53543829>:
>
> > +(defconst julia-hanging-assignment-regexp
> > +  (concat "^[^#\n]+ "
> > +          (regexp-opt '("=" "+=" "-=" "*=" "/=" "//=" "\\=" "^=" "÷=" "%=" "|=" "&=" "$=" "<<=" ">>="
> > +                        ".+=" ".-=" ".*=" "./=" ".//=" ".\\=" ".^=" ".÷=" ".%="))
> > +          (regexp-opt '(" #" " \n" "#" "\n"))))
> > +
> > +(defconst julia-hanging-operator-regexp
> > +  (concat "^[^#\n]+ "
> > +          (regexp-opt '("^" ".^" "//" ".//" "*" "/" "%" "&" "\\" ".*" "./" ".%" ".\\"
> > +                        "<<" ">>" ">>>" ".<<" ".>>" ".>>>"
> > +                        "+" "-" "|" "$" ".+" ".-"
> > +                        ":" ".." "|>"
> > +                        ">" "<" ">=" "<=" "==" "===" "!=" "!==" "<:" "≠" "≤" "≥"
> > +                        ".>" ".<" ".>=" ".<=" ".==" ".!="
> > +                        "&&" "||" "?"))
> > +          (regexp-opt '(" #" " \n" "#" "\n"))))
>
> Thanks for the explaination, I don't really think the current behavior is
> particularly important (since other language modes do not do this). You can
> probably find why it is introduced with git-blame
> <https://github.com/JuliaLang/julia/blame/master/contrib/julia-mode.el>
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/JuliaLang/julia/pull/15156/files#r53543829>.
>